message_app_client/src/state.js

```diff
@@ -3,13 +3,16 @@ import io from 'socket.io-client';
 // When needed, we will pass a custom api URL (test env, prod env) using .env
 const apiURL = process.env.API_URL || 'http://127.0.0.1:5000' ;
 
-export const socket = io(`${apiURL}messages`, { transports: ["websocket"] });
+export const socket = io(`${apiURL}/messages`, { transports: ["websocket"] });
 
-export const getApiRoute = (name) => {
-    return `${apiURL}/${apiRoutes[name]}`;
+export const getApiRoute = (name, parameter=null) => {
+    let URL = `${apiURL}/${apiRoutes[name]}`;
+    if (parameter !== null) URL = URL + `/${parameter}`;
+    return URL;
 }
 export const apiRoutes = {
     signin: "api/signin",
+    currentUserConversations: "api/get_conversations"
```

Comment:
Because we all know that we are getting conversations of the current user, the name could be changed to `getConversations`.